product/tutorials/tutorial_add_fulfillment_places.md

```diff
@@ -0,0 +1,147 @@
+
+Open **product/add_remove_fulfillment.py** and check the AddFulfillmentPlacesRequest.
+
+To add the fulfillment places, open terminal and run the command:
+
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the responses in the Terminal. As you can see, the product was initially created without the fulfillment information, 
+then check the get_product() response, the fulfillment places 'store1', 'store2', 'store3' was added to the fulfillment type 'pickup-in-store'.
+
+## Send out-of-order add_fulfillment_places request
+
+The AddFulfillmentPlaces method allows the caller to specify an update time for when the request is issued.
+This update time is compared against the latest update time recorded for the relevant inventory fields,
+and the update is committed if and only if the update time is strictly after the latest update time.
+
+Let's modify the ```add_fulfillment_request```, change the list of the **place_ids**.
+
+Change the ```request_time``` value, set now() - 1 day:
+```
+request_time = datetime.datetime.now() - datetime.timedelta(days=1)
+```
+
+Run the code once again:
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the product printed out in the Terminal, the fulfillment places information was not updated.
+
+## Remove fulfillment places 
+
+The RemoveFulfillmentPlacesRequest is pretty match the same as AddFulfillmentPlacesRequest,
+
+Check the RemoveFulfillmentPlacesRequest in **product/add_remove_fulfillment.py**, get_remove_fulfillment_request() method.
+
+Next, comment out the lines:
+```
+create_product(product_id)
+add_fulfillment_places(product_name)
+```
+
+Uncomment **```remove_fulfillment_places(product_name)```**.
+
+Run the code sample again with the command:
+
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the responses in the Terminal. The **'store1'**  was remover from the list of fulfillment places for type 'pickup-in-store'.
+
+## Send out-of-order remove_fulfillment_places request
+
+The RemoveFulfillmentPlaces method also allows to set an update time,
+if the value is strictly after the latest update time recorded for the relevant inventory fields, the update will be committed.
+
+Next, modify the ```remove_fulfillment_request```, change the list of the **place_ids**, 
+for example, you can add 'store3' value to the list in ```remove_fulfillment_request.place_ids```.
+
+Set ```request_time``` value to yesterday's date:
+```
+request_time = datetime.datetime.now() - datetime.timedelta(days=1)
+```
+
+Uncomment the line ```delete_product(product_name)``` to clean up after these exercises. 
+
+Run the code one more time:
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the product printed out in the Terminal, the fulfillment places information was not updated.
```

Comment:
Check the product printed out in the Terminal. The fulfillment places information was not updated.